lib/open_api/docs/doc_v3.rb

```diff
@@ -11,37 +11,34 @@ def version
         @version ||= Gem::Version.new(content.fetch_path("info", "version"))
       end
 
-      def definitions
-        @definitions ||= OpenApi::Docs::ComponentCollection.new(self, "definitions")
+      def parameters
+        @parameters ||= OpenApi::Docs::ComponentCollection.new(self, "components/parameters")
       end
 
-      def parameters
-        @parameters ||= OpenApi::Docs::ComponentCollection.new(self, "parameters")
+      def schemas
+        @schemas ||= OpenApi::Docs::ComponentCollection.new(self, "components/schemas")
+      end
+
+      def definitions
+        schemas
       end
 
       def example_attributes(key)
-        definitions[key]["properties"].each_with_object({}) do |(col, stuff), hash|
+        schemas[key]["properties"].each_with_object({}) do |(col, stuff), hash|
           hash[col] = stuff["example"] if stuff.key?("example")
         end
       end
 
       def base_path
-        @base_path ||= @content["basePath"]
+        @base_path ||= @content["servers"].first["variables"]["basePath"]["default"]
       end
 
       def paths
         @content["paths"]
       end
 
       def routes
-        @routes ||= begin
-          paths.flat_map do |path, hash|
-            hash.collect do |verb, _details|
-              p = File.join(base_path, path).gsub(/{\w*}/, ":id")
-              {:path => p, :verb => verb.upcase}
-            end
-          end
-        end
+        @routes ||= Docs.path_routes(base_path, paths)
```

Comment:
I don't think you need this refactor anymore...you did before when you were trying to share v2 and v3 code, but now you don't.